src/main/java/com/direwolf20/buildinggadgets/common/items/pastes/GenericPasteContainer.java

```diff
@@ -0,0 +1,51 @@
+package com.direwolf20.buildinggadgets.common.items.pastes;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextFormatting;
+
+public abstract class GenericPasteContainer extends Item {
+
+    public GenericPasteContainer() {
+        setCreativeTab(BuildingGadgets.BUILDING_CREATIVE_TAB);
+        setMaxStackSize(1);
+    }
+
+    /**
+     * Helper method. Delegates to {@link GenericPasteContainer#setPastes(ItemStack, int)}.
+     */
+    public static void setPasteAmount(ItemStack stack, int amount) {
+        Item item = stack.getItem();
+        if (item instanceof GenericPasteContainer) {
+            ((GenericPasteContainer) item).setPastes(stack, amount);
+        } else {
```

Comment:
unnecessarily nested else clause

Reply:
Do you mean I should remove the log? I cannot figure out how is this unnecessary.

Reply:
Sorry I'm an idiot who swaps lines, never mind